### ISSUE TITLE
Swift 5.5 installation fails

### ISSUE THREAD

**Issue by mattgodbolt** [compiler-explorer/infra#605]
```
2021-10-05 12:58:48,628 lib.installation INFO     Moving from staging (/opt/compiler-explorer/staging/swift-5.5) to final destination (/opt/compiler-explorer/swift-5.5)
/opt/compiler-explorer/swift-5.5/usr/bin/swift-driver: /usr/lib/x86_64-linux-gnu/libcurl.so.4: version `CURL_OPENSSL_3' not found (required by /opt/compiler-explorer/swift-5.5/usr/bin/../lib/swift/linux/libFoundationNetworking.so)
2021-10-05 12:58:50,145 lib.installation ERROR    compilers/swift 5.5 installed OK, but doesn't appear as installed after
0 packages installed OK, 12 skipped, and 1 failed installation
Failed:
  compilers/swift 5.5
```
as noted on the admin node (Ubuntu 20.04.3). CC @Lukasa any ideas? Seems the swift binary has a system dependency on some version of CURL that's not around.

**Comment by Lukasa** [comment]
Interesting, that's a good question. I'll take a bit of a dig and try to work out what's going on there.

**Comment by Lukasa** [comment]
So this looks like we finally ran into trouble when using the old Swift toolchains against the new builder. The swift YAML file calls for Ubuntu 16.04 binaries for Swift:

https://github.com/compiler-explorer/infra/blob/cd944d9a6301a92ef6007ed020c351fdafe31616/bin/yaml/swift.yaml#L10

These naturally are only guaranteed to work on Ubuntu 16.04. So far as I can see the builder now uses 20.04, so we need to grab 20.04 images. However, swift.org only started supporting Ubuntu 20.04 with Swift 5.2.4. In principle this means that building for the old Swift compilers should fail: I don't understand CE's infrastructure well enough to know why this isn't a problem.

In either case, we should be able to just add an `older`/`newer` split to the yaml file and start using the Ubuntu 20.04 images from 5.3.

**Comment by mattgodbolt** [comment]
Oh! Easy fix then! We can install the 20.04 ones. I'll update now!

**Comment by mattgodbolt** [comment]
```
2021-10-05 14:09:34,816 lib.installation INFO     Moving from staging (/opt/compiler-explorer/staging/swift-5.5) to final destination (/opt/compiler-explorer/swift-5.5)                                                                                                                
2021-10-05 14:09:34,829 lib.installation INFO     Destination /opt/compiler-explorer/swift-5.5 exists, temporarily moving out of the way (to /opt/compiler-explorer/staging/temp_orig)                                                                                                  
2021-10-05 14:09:45,243 lib.installation INFO     compilers/swift 5.5 installed OK                                                          
1 packages installed OK, 0 skipped, and 0 failed installation                                                                               
```
